dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/domain/LogSerializer.kt

```diff
@@ -148,6 +154,9 @@ internal class LogSerializer(private val logConstraints: LogConstraints = Datado
         private const val ISO_8601 = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
 
         internal const val TAG_DATADOG_TAGS = "ddtags"
+        internal const val USER_EXTRA_GROUP_VERBOSE_NAME = "user extra information"
+        private const val MAX_ATTRIBUTES_DEPTH_LEVEL = 10
+        private const val MAX_CUSTOM_USER_INFO_DEPTH_LEVEL = 8
```

Comment:
IMHO this is hard to understand for SDK devs and our customers, can't we validate all the attributes (custom user info and others) at once altogether?

Reply:
It would be nice but unfortunately you cannot validate them all together. There are different rules (depths for each group) and for each event type. Also the warning messages should be different. I really don't see what's wrong in this approach as we are using the same method to validate all the attributes. Only the arguments of the method are different. Regarding the depths values the customer don't need to know our internal implementation...they only care about the Public API methods which have the proper docs.

Reply:
I explained below why I chose to do that. It was mostly for warning messages to make them relevant as users will not know about our prefixes and will confuse them.